lib/metrics/metrics-template.c

```diff
@@ -133,7 +63,7 @@ _add_dynamic_labels_vp_helper(const gchar *name, LogMessageValueType type, const
   g_string_append_len(value_buffer, value, value_len);
 
   StatsClusterLabel label = stats_cluster_label(name_buffer->str, value_buffer->str);
-  label_buffers = g_array_append_vals(label_buffers, &label, 1);
+  metrics_tls_cache_append_label(&label);
```

Comment:
This could easily use metrics_tls_cache_get_next_label() instead of initializing a label locally and then copying it.

I don't think this is much of a performance issue as the label itself is small, but this means we have two distinct ways for accessing the label array.

Reply:
Good idea, fixed.